lib/workers/repository/onboarding/branch/onboarding-branch-cache.ts

```diff
@@ -28,3 +42,47 @@ export function deleteOnboardingCache(): void {
     delete cache.onboardingBranchCache;
   }
 }
+
+// checks of onboarding branch has been modified since last run
+export async function hasOnboardingBranchChanged(
+  onboardingBranch: string
+): Promise<boolean> {
+  const cache = getCache();
+  const onboardingSha = getBranchCommit(onboardingBranch);
+  let isModified = false;
+
+  if (cache.onboardingBranchCache) {
+    isModified =
+      onboardingSha !== cache.onboardingBranchCache.onboardingBranchSha;
+  } else {
+    isModified = await scm.isBranchModified(onboardingBranch);
+  }
+
+  return isModified;
+}
```

Comment:
There still seems to be something wrong here. This logic sees wrong:

- If there's already an onboarding cache, then answer whether the onboarding branch has changed since the last run
- If there's no onboarding cache, answer whether the onboarding branch is modified _at all_

These are two different things, and should not be in one function